plugin/vcs/github/github.go

```diff
@@ -0,0 +1,286 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/vcs"
+)
+
+func init() {
+	vcs.Register(vcs.GitHubCom, newProvider)
+}
+
+var _ vcs.Provider = (*Provider)(nil)
+
+// Provider is a GitHub.com VCS provider.
+type Provider struct {
+	l      *zap.Logger
+	client *http.Client
+}
+
+func newProvider(config vcs.ProviderConfig) vcs.Provider {
+	if config.Client == nil {
```

Comment:
Good catch!

Reply:
Init GitLab's Client without actually using it seems weird, but using it makes many more changes out of scope changes for this PR, I'd like to address this in a followup PR instead of this one, including potential reuse of `retry` functions.

Reply:
SG.